build.gradle

```diff
@@ -1,25 +1,28 @@
-apply plugin: 'java'
-apply plugin: 'maven'
-apply plugin: 'eclipse'
-apply plugin: 'idea'
-
+plugins {
+  id 'com.github.ben-manes.versions' version '0.25.0'
+  id 'java'
+  id 'maven'
+  id 'maven-publish'
+  id 'eclipse'
+  id 'idea'
+  id 'com.github.johnrengelman.shadow' version '2.0.4'
```

Comment:
Pretty sure Shadow 5.x is recommended for Gradle 5.x, at least according to the [shadow plugin docs](https://imperceptiblethoughts.com/shadow/changes/#v5-1-0-2019-06-29). I believe one of the bigger advantages is the better compatibility for lazy evaluation of configuration/tasks.